File: frontEnd/react-frontend/src/services/userService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/users';

class UserService {
  // Register new user
  registerUser(userData) {
    return axios.post(`${API_BASE_URL}/signup`, userData);
  }

  // Login user
  loginUser(email, password) {
    return axios.post(`${API_BASE_URL}/login`, { email, password });
  }

  // Get all users
  getAllUsers() {
    return axios.get(API_BASE_URL);
  }

  // Get user by ID
  getUserById(id) {
    return axios.get(`${API_BASE_URL}/${id}`);
  }

  // Delete user
  deleteUser(id) {
    return axios.delete(`${API_BASE_URL}/${id}`);
  }
}

export default new UserService();

File: backend/inventory_app/src/main/java/net/java/inventory_app/service/UserService.java
package net.java.inventory_app.service;

import net.java.inventory_app.entity.User;
import java.util.List;

public interface UserService {
    
    // Register a new user
    User registerUser(User user);
    
    // Login user
    User loginUser(String email, String password);
    
    // Get all users
    List<User> getAllUsers();
    
    // Get user by ID
    User getUserById(Long id);
    
    // Delete user
    void deleteUser(Long id);
}